spec/integration/parser/compiler_spec.rb

```diff
@@ -520,6 +520,15 @@ class foo::bar::baz {
           PP
         }.to raise_error(/Could not find resource 'Notify\[tooth_fairy\]' in parameter 'require'/)
       end
+
+      it 'faulty references are reported with source location' do
+        expect { 
+          compile_to_catalog(<<-PP)
+            notify{ x : 
+              require => tooth_fairy }
+          PP
+        }.to raise_error(/"tooth_fairy" is not a valid resource reference.*\(line: 2\)/)
+      end
```

Comment:
Missing a test for a semantically legal resource reference that isn't found (i.e. when `catalog.resource` returns `nil` instead of raising an `ArgumentError`).

Reply:
How is that relevant to the issue being fixed here - that faulty references were reported without file/line? The reporting of file/line is unchanged for that code path and there are plenty of tests for "Could not find resource ..." - did you mean that there is no test that tests file/line for those?